src/sentry/static/sentry/app/stores/repositoryStore.tsx

```diff
@@ -0,0 +1,88 @@
+import Reflux from 'reflux';
+
+import RepoActions from 'app/actions/repositoryActions';
+import {Repository} from 'app/types';
+
+type RepositoryStoreInterface = {
+  get(
+    orgSlug?: string
+  ): {
+    repos: Array<Repository> | undefined;
+    reposLoading: boolean | undefined;
+    reposError: Error | undefined;
+  };
+
+  orgSlug: string | undefined;
```

Comment:
Would `orgSlug?: string` work for these type definitions?

Reply:
Both works but have slightly different meanings.

* `orgSlug: string | undefined` means that the key will be present, but the value can be `undefined`.
* `orgSlug?: string` means that the key may not be present. 

If we are doing a value check (e.g. `if (props.orgSlug === undefined)`), then there will be no difference in either declarations. 

However, if we are doing something with the keys (e.g. `if (orgSlug in props)`, `Object.keys(props)`), then there's a difference. 

In this case, I'll always have the keys declared in the store, hence the original implementation is more precise.

But... we aren't doing anything with the keys in the store and I can see the improvement in readability, so I've changed it. I'm fine with either way, any thoughts @billyvg @EvanPurkhiser ?

Reply:
Changed my mind about being fine with either implementation.

For the HOC `withRepositories.tsx`, I'm declaring `... | undefined` because I want the child component to have the guarantee that the key will always be there.